src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java

```diff
@@ -254,7 +266,8 @@ private Result executeCmd(String cmd) throws IOException, InterruptedException {
                     .stderr(stderr)
                     .cmdAsSingleString(cmd)
                     .start().join() == 0;
-            String stdoutStr = stdout.toString();
+            String stdoutStr = baos.toString(Computer.currentComputer().getDefaultCharset().name());
```

Comment:
The only reliable way is to call the conversion method on the slave instead of the master. So you may need to launch a more complex call on the slave's side